metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_api.py

```diff
@@ -0,0 +1,706 @@
+                continue
+
+            column_dictionary[table_full_path].append(
+                {
+                    "name": record["COLUMN_NAME"],
+                    "ordinal_position": record.get(
+                        "ORDINAL_POSITION", ordinal_position
+                    ),
+                    "is_nullable": record["IS_NULLABLE"],
+                    "data_type": record["DATA_TYPE"],
+                    "column_size": record["COLUMN_SIZE"],
+                }
+            )
+
+            ordinal_position += 1
+
+            if record.get("TABLE_SCHEMA") not in schema_list:
+                schema_list.append(record.get("TABLE_SCHEMA"))
+
+        distinct_tables_list = list(
+            {
+                tuple(
+                    dictionary[key]
+                    for key in (
+                        "TABLE_SCHEMA",
+                        "TABLE_NAME",
+                        "FULL_TABLE_PATH",
+                        "VIEW_DEFINITION",
+                    )
+                    if key in dictionary
+                ): dictionary
+                for dictionary in tables_and_columns
+            }.values()
+        )
+
+        for schema in schema_list:
+            schema_dict_lookup.append(self.validate_schema_format(schema))
+
+        for table, schemas in product(distinct_tables_list, schema_dict_lookup):
+            if table.get("TABLE_SCHEMA") == schemas.get("original_path"):
+                dataset_list.append(
+                    {
+                        "TABLE_SCHEMA": "["
+                        + ", ".join(
+                            schemas.get("formatted_path") + [table.get("TABLE_NAME")]
+                        )
+                        + "]",
+                        "TABLE_NAME": table.get("TABLE_NAME"),
+                        "COLUMNS": column_dictionary.get(
+                            table.get("FULL_TABLE_PATH", "")
+                        ),
+                        "VIEW_DEFINITION": table.get("VIEW_DEFINITION"),
+                        "RESOURCE_ID": self.get_dataset_id(
+                            schema=".".join(schemas.get("formatted_path")),
+                            dataset=table.get("TABLE_NAME", ""),
+                        ),
+                        "LOCATION_ID": self.get_dataset_id(
+                            schema=".".join(schemas.get("formatted_path")),
+                            dataset="",
+                        ),
+                    }
+                )
+
+        return dataset_list
+
+    def get_all_tables_and_columns(self, containers: Deque) -> List[Dict]:
+        if self.edition == DremioEdition.ENTERPRISE:
+            query_template = DremioSQLQueries.QUERY_DATASETS_EE
+        elif self.edition == DremioEdition.CLOUD:
+            query_template = DremioSQLQueries.QUERY_DATASETS_CLOUD
+        else:
+            query_template = DremioSQLQueries.QUERY_DATASETS_CE
+
+        def get_pattern_condition(
```

Comment:
Nested methods are not recommended. Can we move this outside or at module

Reply:
Changed